File: logsize/decibyte.hs
{-# LANGUAGE OverloadedStrings #-}

import System.Environment
import System.Posix.Files
import Text.Printf

data DecibyteFile = DecibyteFile {
  name :: String,
  size :: Integer 
} deriving (Show)

-- log10(n) * 10
log10size :: Integer -> Double
log10size size = (logBase 10 (fromIntegral size)) * 10

-- Get the metadata for a specific file
stat_file :: String -> IO DecibyteFile
stat_file str = do
    stat <- getFileStatus str
    let size = toInteger (fileSize stat)
    return (DecibyteFile str size)

decibyte_files :: [String] -> Integer -> IO () 
decibyte_files xs i
  | null xs = printf "%9.2fdBy === TOTAL ===\n" (log10size i)
  | otherwise = do
    stat <- stat_file (head xs)
    printf "%9.2fdBy %s\n" (log10size (size stat)) (name stat)
    decibyte_files (tail xs) (i+(size stat))

main :: IO ()
main = do
  a <- getArgs;
  decibyte_files a 0

File: logsize/decibyte.py
import os, sys, math

def filesize(f):
    filesize = os.stat(f).st_size
    return (f, filesize)

def decibyte(bytelen):
    return math.log10(bytelen)*10

def main():
    total = 0.0
    files = map(filesize, sys.argv[1:])
    try:
        for f in files:
            total += f[1]
            print("{:9.2f}dBy {}".format(decibyte(f[1]), f[0]))
    except FileNotFoundError as e:
        print("{}".format(e))
        sys.exit(2)
    print("{:9.2f}dBy === TOTAL ===".format(decibyte(total)))

if __name__ == '__main__':
    main()
